scipy/sparse/array_api/csr.py

```diff
@@ -0,0 +1,117 @@
+from collections import namedtuple
+from scipy import sparse
+import numpy as np
+import numpy.typing as npt
+
+def asarray(obj, *, dtype=None, copy=None):
+    # This is a HUGE hack to get the array-api test suite even running, much less doing anything sensible.
+    if np.isscalar(obj):
+         obj = np.array(obj).reshape(1, -1)
+    if isinstance(obj, list):
+        obj = np.array(obj).reshape(len(obj), -1)
+    return sparse.csr_array(obj, dtype=dtype, copy=copy)
+
+def empty(shape, *, dtype=None):
+    return sparse.csr_array(shape, dtype=dtype)
+
+def empty_like(x, *, dtype=None):
+    return sparse.csr_array(x, dtype=dtype, copy=False)
+
+def eye(n_rows, n_cols=None, /, *, k=0, dtype=None):
+    return sparse.eye(n_rows, n_cols, k=k, dtype=dtype, format="csr")
+
+def full(shape, fill_value, *, dtype=None):
+    return sparse.csr_array(np.full(shape, fill_value, dtype=dtype))
+
+def full_like(x, fill_value, *, dtype=None):
+    return sparse.csr_array(np.full_like(x, fill_value, dtype=dtype))
+
+def ones(shape, *, dtype=None):
+    return sparse.csr_array(np.ones(shape, dtype=dtype))
+
+def ones_like(x, *, dtype=None):
+    return sparse.csr_array(np.ones_like(x, dtype=dtype))
+
+def zeros(shape, *, dtype=None):
+    if isinstance(shape, int):
+        return sparse.csr_array((shape,1), dtype=dtype)
+    return sparse.csr_array(shape, dtype=dtype)
+
+def zeros_like(x, *, dtype=None):
+    if dtype is None:
+        dtype = x.dtype
+    return sparse.csr_array(x.shape, dtype=dtype)
+
+def floor(x):
+    return sparse.csr_array((np.floor(x.data), x.indices, x.indptr), shape=x.shape)
+
+def take(x, indices, /, *, axis: int | None = None):
+    idxer = [slice(None)] * x.ndim
+    idxer[axis] = indices
+    return x[tuple(idxer)]
+
+def all(x, /, *, axis: int | tuple[int, ...] | None = None, keepdims: bool = False): # type: ignore
+    return np.sum(x, axis=axis)
+
+def isnan(x):
```

Comment:
Signature should use positional-only:
```python
def isnan(x, /):
```

This applies also to other functions where `x` is the only argument.